runtime/compiler/z/codegen/S390AOTRelocation.cpp

```diff
@@ -260,7 +260,7 @@ void TR::S390EncodingRelocation::addRelocation(TR::CodeGenerator *cg, uint8_t *c
          line,
          node);
       }
-   else if (_reloType==TR_MethodTypeTableEntryAddress)
+   else if (comp->compileRelocatableCode() && _reloType==TR_MethodTypeTableEntryAddress)
```

Comment:
If we reach here with `TR_CallsiteTableEntryAddress` or `TR_MethodTypeTableEntryAddress` in a non-AOT compilation, then we'll go to the unhandled relo type assertion, which is probably not what we want (even though `TR_ASSERT` is almost always compiled out)

Reply:
😞 

Updated in force push.